install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -274,7 +274,7 @@ data:
 
 
 
-{{- if hasKey .Values.bpf "mapDynamicSizeRatio" }}
+{{- if .Values.bpf.mapDynamicSizeRatio }}
   # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
   # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
   bpf-map-dynamic-size-ratio: {{ .Values.bpf.mapDynamicSizeRatio | quote }}
```

Comment:
From the comment I'm guessing that `0.0` is a valid configuration value, but my understanding is that after the patch it would be ignored. Is it a fair statement?

Reply:
what does `0.0` ratio do? no system memory for dynamic sizing? I could look into the code to see what `0.0` could do.

Reply:
based on the code, I don't think  `dynamicSizeRatio` with `0.0` is used in the code, doesn't make sense with `0.0`, so we should correct the statement that dynamicSizeRatio should be `0.0 < dynamicSizeRatio <= 1.0`

```
        // Allow the range (0.0, 1.0] because the dynamic size will anyway be
        // clamped to the table limits. Thus, a ratio of e.g. 0.98 will not lead
        // to 98% of the total memory being allocated for BPF maps.
        dynamicSizeRatio := vp.GetFloat64(MapEntriesGlobalDynamicSizeRatioName)
        if 0.0 < dynamicSizeRatio && dynamicSizeRatio <= 1.0 {
                vms, err := mem.VirtualMemory()
                if err != nil || vms == nil {
                        log.WithError(err).Fatal("Failed to get system memory")
                }
                c.calculateDynamicBPFMapSizes(vp, vms.Total, dynamicSizeRatio)
                c.BPFMapsDynamicSizeRatio = dynamicSizeRatio
        } else if dynamicSizeRatio < 0.0 {
                return fmt.Errorf("specified dynamic map size ratio %f must be ≥ 0.0", dynamicSizeRatio)
        } else if dynamicSizeRatio > 1.0 {
                return fmt.Errorf("specified dynamic map size ratio %f must be ≤ 1.0", dynamicSizeRatio)
        }       
```

Reply:
so we have default dynamicSizeRatio set to `0.0` if version < 1.8, but still not used

```
{{- $defaultBpfMapDynamicSizeRatio := 0.0 -}}

{{- /* Default values when 1.8 was initially deployed */ -}}
{{- if semverCompare ">=1.8" (default "1.8" .Values.upgradeCompatibility) -}}

  {{- $defaultBpfMapDynamicSizeRatio = 0.0025 -}}

{{- if .Values.bpf.mapDynamicSizeRatio }}
  # Specifies the ratio (0.0-1.0] of total system memory to use for dynamic
  # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
  bpf-map-dynamic-size-ratio: {{ .Values.bpf.mapDynamicSizeRatio | quote }}
{{- else if ne $defaultBpfMapDynamicSizeRatio 0.0 }}
  # Specifies the ratio (0.0-1.0) of total system memory to use for dynamic
  # sizing of the TCP CT, non-TCP CT, NAT and policy BPF maps.
  bpf-map-dynamic-size-ratio: {{ $defaultBpfMapDynamicSizeRatio | quote }}
{{- end }}

```

Reply:
Thanks for investigating!